src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,496 @@
+  0 ≤[E] μ.to_signed_measure - (ε • ν).to_signed_measure :=
+begin
+  -- for all `n : ℕ`, obtain the Hahn decomposition for `μ - (1 / n) ν`
+  have : ∀ n : ℕ, ∃ i : set α, measurable_set i ∧
+    0 ≤[i] (μ.to_signed_measure - ((1 / (n + 1) : ℝ≥0) • ν).to_signed_measure) ∧
+    (μ.to_signed_measure - ((1 / (n + 1) : ℝ≥0) • ν).to_signed_measure) ≤[iᶜ] 0,
+  { intro, exact exists_compl_positive_negative _ },
+  choose f hf₁ hf₂ hf₃ using this,
+  -- set `A` to be the intersection of all the negative parts of obtained Hahn decompositions
+  -- and we show that `μ A = 0`
+  set A := ⋂ n, (f n)ᶜ with hA₁,
+  have hAmeas : measurable_set A,
+  { exact measurable_set.Inter (λ n, (hf₁ n).compl) },
+  have hA₂ : ∀ n : ℕ, (μ.to_signed_measure - ((1 / (n + 1) : ℝ≥0) • ν).to_signed_measure) ≤[A] 0,
+  { intro n, exact restrict_le_restrict_subset _ _ (hf₁ n).compl (hf₃ n) (set.Inter_subset _ _) },
+  have hA₃ : ∀ n : ℕ, μ A ≤ (1 / (n + 1) : ℝ≥0) * ν A,
+  { intro n,
+    have := nonpos_of_restrict_le_zero _ (hA₂ n),
+    rwa [to_signed_measure_sub_apply hAmeas, sub_nonpos, ennreal.to_real_le_to_real] at this,
+    exacts [ne_of_lt (measure_lt_top _ _), ne_of_lt (measure_lt_top _ _)] },
+  have hμ : μ A = 0,
+  { lift μ A to ℝ≥0 using ne_of_lt (measure_lt_top _ _) with μA,
+    lift ν A to ℝ≥0 using ne_of_lt (measure_lt_top _ _) with νA,
+    rw ennreal.coe_eq_zero,
+    by_cases hb : 0 < νA,
+    { suffices : ∀ b, 0 < b → μA ≤ b,
+      { by_contra,
+        have h' := this (μA / 2) (nnreal.half_pos (zero_lt_iff.2 h)),
+        rw ← @not_not (μA ≤ μA / 2) at h',
+        exact h' (not_le.2 (nnreal.half_lt_self h)) },
+      intros c hc,
+      have : ∃ n : ℕ, 1 / (n + 1 : ℝ) < c * νA⁻¹, refine exists_nat_one_div_lt _,
+      { refine mul_pos hc _,
+        rw _root_.inv_pos, exact hb },
+      rcases this with ⟨n, hn⟩,
+      have hb₁ : (0 : ℝ) < νA⁻¹, { rw _root_.inv_pos, exact hb },
+      have h' : 1 / (↑n + 1) * νA < c,
+      { rw [← nnreal.coe_lt_coe, ← mul_lt_mul_right hb₁, nnreal.coe_mul, mul_assoc,
+            ← nnreal.coe_inv, ← nnreal.coe_mul, _root_.mul_inv_cancel, ← nnreal.coe_mul,
+            mul_one, nnreal.coe_inv],
+        { convert hn, simp },
+        { exact ne.symm (ne_of_lt hb) } },
+      refine le_trans _ (le_of_lt h'),
+      rw [← ennreal.coe_le_coe, ennreal.coe_mul],
+      exact hA₃ n },
+    { rw [not_lt, le_zero_iff] at hb,
+      specialize hA₃ 0,
+      simp [hb, le_zero_iff] at hA₃,
+      assumption } },
+  -- since `μ` and `ν` are not mutually singular, `μ A = 0` implies `ν Aᶜ > 0`
+  rw mutually_singular at h, push_neg at h,
+  have := h _ hAmeas hμ,
+  simp_rw [hA₁, set.compl_Inter, compl_compl] at this,
+  -- as `Aᶜ = ⋃ n, f n`, `ν Aᶜ > 0` implies there exists some `n` such that `ν (f n) > 0`
+  obtain ⟨n, hn⟩ := exists_measure_pos_of_not_measure_Union_null this,
+  -- thus, choosing `f n` as the set `E` suffices
+  exact ⟨1 / (n + 1), by simp, f n, hf₁ n, hn, hf₂ n⟩,
+end
+
+section
```

Comment:
In this section, you use definitions which are quite specific, but their name is quite general. Same thing for several lemmas below. For all the statements in the proof, do you think it is worth working in a dedicated namespace like `measure_theory.measure.lebesgue_decomposition`?
```suggestion
namespace lebesgue_decomposition
```

Reply:
good idea!